src/app/core/data/root-data.service.ts

```diff
@@ -0,0 +1,72 @@
+import { DataService } from './data.service';
+import { Root } from './root.model';
+import { Injectable } from '@angular/core';
+import { ROOT } from './root.resource-type';
+import { dataService } from '../cache/builders/build-decorators';
+import { RequestService } from './request.service';
+import { RemoteDataBuildService } from '../cache/builders/remote-data-build.service';
+import { Store } from '@ngrx/store';
+import { CoreState } from '../core.reducers';
+import { ObjectCacheService } from '../cache/object-cache.service';
+import { HALEndpointService } from '../shared/hal-endpoint.service';
+import { NotificationsService } from '../../shared/notifications/notifications.service';
+import { HttpClient } from '@angular/common/http';
+import { DefaultChangeAnalyzer } from './default-change-analyzer.service';
+import { Observable } from 'rxjs';
+import { RemoteData } from './remote-data';
+import { FollowLinkConfig } from '../../shared/utils/follow-link-config.model';
+
+/* tslint:disable:max-classes-per-file */
+
+/**
+ * A private DataService implementation to delegate specific methods to.
+ */
+class DataServiceImpl extends DataService<Root> {
+  protected linkPath = '';
+  protected responseMsToLive = 6 * 60 * 60 * 1000;
+
+  constructor(
+    protected requestService: RequestService,
+    protected rdbService: RemoteDataBuildService,
+    protected store: Store<CoreState>,
+    protected objectCache: ObjectCacheService,
+    protected halService: HALEndpointService,
+    protected notificationsService: NotificationsService,
+    protected http: HttpClient,
+    protected comparator: DefaultChangeAnalyzer<Root>) {
+    super();
+  }
+}
+
+/**
+ * A service to retrieve the {@link Root} object from the REST API.
+ */
+@Injectable()
+@dataService(ROOT)
+export class RootDataService {
```

Comment:
Any class with the `@dataService` annotation should implement at least findHref and findAllByHref, even if you use delegation. Otherwise linkService can crash at runtime. There's no type checking for that at the moment. https://github.com/DSpace/dspace-angular/issues/742 will address that